heka/sandbox/decoders/extract_telemetry_dimensions.lua

```diff
@@ -93,120 +106,117 @@ function parse_creation_date(date)
-        end
+        if type(app) ~= "table" then return "missing application object" end
+
+        local cts = parse_creation_date(parsed.creationDate)
+        if not cts then return "missing creationDate"end
+        update_field(msg.Fields, "creationTimestamp", cts)
 
         if type(parsed.payload) == "table" and
            type(parsed.payload.info) == "table" then
-            msg.Fields.reason = parsed.payload.info.reason
-        end
-
-        msg.Fields.creationTimestamp = parse_creation_date(parsed.creationDate)
-        if not msg.Fields.creationTimestamp then
-           return -1, "missing creationDate"
+               update_field(msg.Fields, "reason", parsed.payload.info.reason)
         end
 
         if type(parsed.environment) == "table" and
            type(parsed.environment.system) == "table" and
            type(parsed.environment.system.os) == "table" then
-            msg.Fields.os = parsed.environment.system.os.name
+               update_field(msg.Fields, "os", parsed.environment.system.os.name)
         end
 
-        msg.Fields.sourceVersion    = tostring(parsed.version)
-        msg.Fields.docType          = parsed.type or UNK_DIM
+        update_field(msg.Fields, "sourceVersion", tostring(parsed.version))
+        update_field(msg.Fields, "docType"      , parsed.type or UNK_DIM)
 
         -- Get some more dimensions.
-        msg.Fields.appName          = app.name or UNK_DIM
-        msg.Fields.appVersion       = app.version or UNK_DIM
-        msg.Fields.appUpdateChannel = app.channel or UNK_DIM
+        update_field(msg.Fields, "appName"           , app.name or UNK_DIM)
+        update_field(msg.Fields, "appVersion"        , app.version or UNK_DIM)
+        update_field(msg.Fields, "appUpdateChannel"  , app.channel or UNK_DIM)
 
         -- Do not want default values for these.
-        msg.Fields.appBuildId       = app.buildId
-        msg.Fields.appVendor        = app.vendor
-        msg.Fields.clientId         = parsed.clientId
+        update_field(msg.Fields, "appBuildId", app.buildId)
+        update_field(msg.Fields, "appVendor" , app.vendor)
+        clientId = parsed.clientId
+        update_field(msg.Fields, "clientId"  , clientId)
 
         -- restructure the main ping message
         if parsed.type == "main" then
-            split_objects(parsed.environment, "environment", environment_objects)
-            split_objects(parsed.payload, "payload", main_ping_objects)
+            split_objects(msg.Fields, parsed.environment, "environment", environment_objects)
+            split_objects(msg.Fields, parsed.payload, "payload", main_ping_objects)
             local ok, json = pcall(cjson.encode, parsed) -- re-encode the remaining data
-            if not ok then return -1, json end
+            if not ok then return json end
             msg.Payload = json
         else
             msg.Payload = json
         end
     end
+    update_field(msg.Fields, "sampleId", sample(clientId, 100))
```

Comment:
Let's explicitly return `nil` with a comment indicating that this is the success case.